src/skottie-player.ts

```diff
@@ -104,26 +130,299 @@ export class SkottiePlayer extends LitElement {
   @property({type: Number})
   height = 0;
 
+  /**
+   * Indicator for file upload.
+   */
+  @property({type: Boolean})
+  fileUploaded = false;
+
+  /**
+   * Indicator for showing JSON editor.
+   */
+  @property({type: Boolean})
+  showJsonEditor = false;
+
+  /**
+   * The content of the uploaded file as a formatted string.
+   */
+  @property({type: String})
+  fileString = '';
+
+  /**
+   * The JSONEditor.
+   */
+  @property({type: JSONEditor})
```

Comment:
Some of these property fields don't seem to be accessed in the html or the passed through attributes. If you only use a class field in the typescript code you can set the field as private without the @property decorator. You should only really decorate fields if you want the html to update when the field is changed, or if the field can be set externally when the element is set in another's html

Typescript private fields: https://www.typescriptlang.org/docs/handbook/classes.html#understanding-typescripts-private
Lit Element property decorator: https://lit-element.polymer-project.org/guide/properties

Reply:
Also the property type only supports String, Number, Boolean, Array, and Object, if you want to use another type you need to use the converter field